content/en/docs/04/labs/42-k8s.md

```diff
@@ -0,0 +1,119 @@
+
+Create a Kubernetes job that pushes the specified metric to the push gateway. Make sure to replace the pushgateway with the url received above.
+
+```bash
+oc -n $team-monitoring create job --image=quay.balgroupit.com/library/toolkit:ubuntu-20.04 pushgw-example1 -- \
+bash -c "echo 'prometheus_training_labs_completed_total 4' | curl --data-binary @- http://pushgateway.[examples]-monitoring:9091/metrics/job/prometheus_training"
+```
+
+{{% alert title="Command Explanation" color="primary" %}}
+If you are not very familiar with `oc create job`. The above command does the following:
+
+* `oc -n ... create job` creates an adhoc [kubernetes job](https://kubernetes.io/docs/concepts/workloads/controllers/job/)
+* `--image=` specifies, which image the container will use. We will use the toolkit container because it provides bash and curl.
+* `pushgw-example1` is the name of the job
+* `bash -c "..."` is the command, the job should execute
+
+{{% /alert %}}
+
+FIXME web UI
+Verify the metric in the [Prometheus we UI](http://LOCALHOST:9090/graph?g0.range_input=1h&g0.expr=prometheus_training_labs_completed_total&g0.tab=1). It may take up to 30s ( Depending on the `scrape_interval`) to be available in Prometheus.
+
+[Push](https://github.com/prometheus/pushgateway/blob/master/README.md#command-line) the following metric (notice the `instance` label) to the Pushgateway and make sure the metric gets scraped by Prometheus
+
+```promql
+# TYPE some_metric_total counter
+# HELP This is just an example metric.
+some_metric_total{job="prometheus_training",instance="myinstance"} 42
+```
+
+{{% details title="Hints" mode-switcher="normalexpertmode" %}}
+
+To push a metric to the Pushgateway, which then will be scraped by Prometheus, we can simply create the following job. Note the actual content of the HTTP request, is just simply the exact metric we want Prometheus to scrape.
+
+Execute the following command to push the metric to your Pushgateway
+
+```bash
+oc -n $team-monitoring create job --image=quay.balgroupit.com/library/toolkit:ubuntu-20.04 pushgw-example2 -- \
+bash -c "cat <<EOF | curl --data-binary @- http://pushgateway.[examples]-monitoring:9091/metrics/job/prometheus_training/instance/myinstance
+# TYPE some_metric_total counter
+# HELP This is just an example metric.
+some_metric_total 42
+EOF"
+```
+
+{{% alert title="Command Explanation" color="primary" %}}
+If you are not very familiar with the Linux shell. The above command does the following:
+
+* the `cat` command reads the actual metric and pipes it to `stdin`
+* curl sends a HTTP POST request to the url <http://localhost:9091/metrics/job/prometheus_training/instance/myinstance> with the --data-binary parameter set to `stdin` (the actual metric)
+
+{{% /alert %}}
+
+FIXME web UI
+Verify the metric in the [Prometheus we UI](http://LOCALHOST:9090/graph?g0.range_input=1h&g0.expr=some_metric_total&g0.tab=1). It may take up to 30s ( Depending on the `scrape_interval`) to be available in Prometheus.
```

Comment:
```suggestion
Verify the metric in the [Prometheus we UI](http://LOCALHOST:9090/graph?g0.range_input=1h&g0.expr=some_metric_total&g0.tab=1). It may take up to 30s (depending on the `scrape_interval`) to be available in Prometheus.
```